### DISCUSSION TITLE
Valhalla QGIS Plugin

### DISCUSSION THREAD

**Discussion by nilsnolde** [valhalla/valhalla#2737] [Show and tell]
Let's try discussions.. 

We developed and released a QGIS plugin to the public repository:
https://plugins.qgis.org/plugins/valhalla/

Some notes on usage etc:
https://gis-ops.com/valhalla-qgis-plugin

Feedback of course welcome! Not everything is implemented (everything is quite a lot in valhalla's case.. :heart: ).

EDIT: code is here btw: https://github.com/gis-ops/valhalla-qgis-plugin (was a testing tool until yesterday, code style is ehh.. PR's still wanted though)

**Comment by danpat** [comment]
I think the biggest question about adding it to the main repo is how do we ensure that it keeps working?  What steps can we add to CI to ensure that a) the Valhalla interface the plugin depends on doesn't change without being noticed, and b) that the plugin functions with QGIS as expected.

**Reply by nilsnolde** [reply]
Ah no oops, misunderstanding @danpat , that wasn't my intention! I actually think this shouldn't be part of this repo, for the minimum reasons you mentioned and then some.

I mostly wanted to advertise it a little:)

**Comment by kevinkreiser** [comment]
While I also think it should continue to live in its own repo I'm actually not worried about breaking the plugin. The plugin itself uses the json API that the router provides, we are very careful to make changes to that in ways that are not breaking clients (additive). So long as the qgis client doesnt barf when it sees new values, everything should be ok :smile:
